airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/cursor.py

```diff
@@ -82,9 +95,21 @@ def __init__(
         self._cursor_field = cursor_field
         # To see some example where the slice boundaries might not be defined, check https://github.com/airbytehq/airbyte/blob/1ce84d6396e446e1ac2377362446e3fb94509461/airbyte-integrations/connectors/source-stripe/source_stripe/streams.py#L363-L379
         self._slice_boundary_fields = slice_boundary_fields if slice_boundary_fields else tuple()
+        self._start = start
         self._most_recent_record: Optional[Record] = None
         self._has_closed_at_least_one_slice = False
-        self.state = stream_state
+        self.start, self._concurrent_state = self._get_concurrent_state(stream_state)
+
+    @property
+    def state(self) -> MutableMapping[str, Any]:
```

Comment:
Should this be private? From a very quick look, I don't see this used outside of the class (which I think is very nice as it means we don't expose non-domain object like `MutableMapping`

Reply:
It will actually be used externally, as you'll see in the follow up Salesforce PR.

Reply:
Can you show me where? I can't see to identify this change on https://github.com/airbytehq/airbyte/pull/33522/files

Reply:
Sorry that was out of date. It's [there](https://github.com/airbytehq/airbyte/pull/33522/files#diff-1180da547d1ffa8dbd6c895174d912b0023e3ccfce51920fab55913a0629baa8R225) now.